### PR TITLE
TR_188 Incomplete Form Alert

### PR DESCRIPTION
**Pull request by bahayg**
# [TR_188] Incomplete Form Alert

---
#### What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)
- Add a component (IncompleteFormAlert)

#### What is the current behavior? (You can also link to an open issue here)
- When users click to a back icon, they are redirected to previous screen without any warning. 

#### What is the new behavior? (if this is a feature change)
- Incomplete Form Alert modal is created to pop up when a user tries to go back before form filling is completed and it warns the users that they will lose the information they put in the form if they go back.

- This alert modal is ready to be used in any form with back action such as create donation and registration.  

#### Other information

- Incomplete Form Alert Modal can be called with the following function:

```
import useGlobal from "@state";

const [ state, { updateAlert } ] = useGlobal() as any;

updateAlert(
   { 
      title: 'ARE YOU SURE?',
      message: 'Test Message', 
      dismissable: false,
   }
);
```

#### Images (before/ after screenshots, interaction GIFs, ...)
- It requires user acknowledgement (undismissable)

Screen Shot 2020-06-12 at 5 57 00 PM

188_resized

---

### CODE REVIEW

File: src/elements/IncompleteFormAlert/IncompleteFormAlert.tsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+							style={{
+								width: 120,
+							}}
+							buttonStyle={{
+								default: scheme.primary,
+								pressed: scheme.secondary,
+								disabled: scheme.disabled,
+							}}
+							onPress={handleCloseButtonPress}
+						/>
```
Comment (emarti43) [inline src/elements/IncompleteFormAlert/IncompleteFormAlert.tsx]: The `YES, CANCEL` button and the `NO, KEEP IT` buttons both trigger the same function. Would these buttons have different behaviors? For instance, `NO KEEP IT` would do nothing except dismiss, and `YES CANCEL` would exit the form?

Comment (bahayg) [reply src/elements/IncompleteFormAlert/IncompleteFormAlert.tsx]: Thank you Edgar for taking your time to review this. 
Yes, you are right, the "YES, CANCEL" button will exit the form. The thing is this modal will be used in many different forms and I did not know how to write that generic function I mean where to take our user. Please let me know if you can help me with this.

Comment (emarti43) [reply src/elements/IncompleteFormAlert/IncompleteFormAlert.tsx]: I'd be happy help!

**Review by emarti43 — APPROVED** [review]
👍

**Pull request merged**